### PR TITLE
feat: support fused moe

### PR DESCRIPTION
**Pull request by jimoosciuc**
## Motivation

- Provide a TPU-friendly, high-performance Mixture-of-Experts (MoE) implementation by fusing routing/top-k, expert FFN compute, cross-device communication, and aggregation into a single Pallas kernel to reduce kernel launches and HBM traffic.
- Support production MoE variants (e.g., grouped top-k, shared expert, optional bias/renorm/scaling, optional sub-channel quantization).

## Modifications

- Implement a fused MoE kernel (fused_ep_moe) and a reference implementation (ref_moe) for correctness comparisons.
Introduce a block configuration object with runtime override + validation rules for tiling/scratch constraints (including token tiling, shared-expert tiling, and sub-channel quant constraints).
- Add feature support for silu/gelu/swigluoai activations, grouped top-k, balanced top-k, router bias, renormalization, and an optional routed scaling factor.
- Add an optional shared-expert path and optional sub-channel quantization support (with associated per-sub-channel scale tensors).
- Add an offline-tuned block-config lookup table keyed by device kind and simplified shape signatures, with a safe default fallback.
- Add benchmarking utilities and expand correctness tests to cover key variants (bias/shared expert/quantization) against the reference path.

## Accuracy Tests
```
evalscope eval  --model inclusionAI/Ling-1T --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets gsm8k --eval-batch-size 128
```
image

## tune block config
```
python -m benchmark.moe.bench_fused_moe --use-shared-expert  --use-grouped-topk --num-tokens 128 --imbalance-mode sparse_hotspot --hotspot-ratio 1 --hotspot-count 48 --tune-block-config
```

## Benchmarking and Profiling
```
python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 512 --random-input 4096 --random-output 1024 --max-concurrency 128 --random-range-ratio 1 --warmup-requests 0
```
image

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [x] (Optional) The necessary documentation update.

### CODE REVIEW

File: python/sgl_jax/srt/kernels/fused_moe/v1/kernel.py [hunk 1]
```diff
@@ -0,0 +1,3092 @@
...
+    ### Semaphores:
+    token_stage_x2_sems,  # DMA(2,): <token_buf_id>
+    acc_stage_x3_sems,  # DMA(3,): <acc_buf_id>
+    local_sems,  # DMA(2, N): weight ping-pong semaphores (plus gating/output and shared-expert staging)
+    send_x2_sems,  # <e_sem_id> (2,)
+    recv_x2_sems,  # <e_sem_id> (2,)
+    a2a_gather_sem,
+    a2a_acc_sems,  # DMA(1,)
+    barrier_sem,
+    expert_size_sem,
```
Comment (Prayer3th) [inline python/sgl_jax/srt/kernels/fused_moe/v1/kernel.py]: Perhaps we should remove it

**Review by Iamleos — APPROVED** [review]
LGTM
